src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppInstanceStatus.java

```diff
@@ -189,4 +150,203 @@ else if (containerStatus.getRestartCount() == 0 &&
 		}
 		return result;
 	}
+
+	static interface RunningPhaseDeploymentStateResolver {
```

Comment:
@dturanski would it make sense to move this interface to a public package? And does it make sense to have the Resolver type to be writable via a setter? Maybe not now, but couldn't we have different strategies that could be chosen and set according to environment? Just asking, seems like a candidate to allow users to change behavior

Reply:
I'll take another look at it